pinot-broker/src/main/java/com/linkedin/pinot/broker/routing/builder/DefaultOfflineRoutingTableBuilder.java

```diff
@@ -28,24 +29,19 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.linkedin.pinot.broker.routing.RoutingTableLookupRequest;
-import com.linkedin.pinot.broker.routing.ServerToSegmentSetMap;
-import com.linkedin.pinot.common.config.TableConfig;
-import com.linkedin.pinot.common.response.ServerInstance;
-import com.linkedin.pinot.transport.common.SegmentIdSet;
 
 /**
  * Create a given number of routing tables based on random selections from ExternalView.
  */
-public class DefaultOfflineRoutingTableBuilder extends AbstractRoutingTableBuilder {
+public class DefaultOfflineRoutingTableBuilder extends BaseRoutingTableBuilder {
   private static final Logger LOGGER = LoggerFactory.getLogger(DefaultOfflineRoutingTableBuilder.class);
+
   private RoutingTableBuilder _largeClusterRoutingTableBuilder;
   private RoutingTableBuilder _smallClusterRoutingTableBuilder;
+  private volatile RoutingTableBuilder _routingTableBuilder;
```

Comment:
Can you explain the logic behind of putting `volatile` for this variable?

Reply:
Added comment